src/huggingface_hub/repository.py

```diff
@@ -266,6 +266,7 @@ def __init__(
         local_dir: str,
         clone_from: Optional[str] = None,
         repo_type: Optional[str] = None,
+        repo_visibility: Optional[str] = None,
```

Comment:
Just happene d to have notifications for this repo.

Isn't `repo_visibility: str = "public"` much better than `Optional[str]` ? Strings are immutable so it's fine I think.

Reply:
why not use an optional bool instead?

Reply:
(and name it the same as in `api.create_repo`)

Reply:
My intention was to have a similar UX and API to what you have when using the UI. There we have repository visibility with `"public"` & `"private"`. But not a strong opinion on this and happy to convert to the same as in `api.create_repo`.

Reply:
i'll defer to @LysandreJik and/or @osanseviero but personally i think it's better to have a bool

Reply:
I think `private: Optional[bool] = None` would be better. That way we avoid having to change from string to bool and we're consistent with `.create_repo`.